docs/development/how-to/ide-setup/intellij-setup.md

```diff
@@ -2,45 +2,56 @@
-  1. *Google Java Format* [plugin](https://plugins.jetbrains.com/plugin/8527-google-java-format)
-        1. **Settings* > google-java-format Settings**
-        1.  Check 'enable'
+  1. *[Google Java Format Plugin](https://plugins.jetbrains.com/plugin/8527-google-java-format)*
+        1. **Settings > Other Settings > google-java-format Settings**
+        1.  Check "Enable google-java-format"
       ![Screenshot from 2019-08-08 17-35-52
       ](https://user-images.githubusercontent.com/12397753/62746114-07cc2b80-ba03-11e9-9ac0-0b1e6e1e8788.png)
-  1. *checkstyle-IDEA* [plugin](https://github.com/jshiell/checkstyle-idea)
-        1. after install finish configuration in: **Other Settings > Checkstyle**
-            1. load checkstyle file by clicking on the "plus" and navigating to the file .\IdeaProjects\triplea\config\checkstyle (If you can't find it, you can download it from [the repository](https://github.com/triplea-game/triplea/blob/master/config/checkstyle/checkstyle.xml))
-            1. set checkstyle version
-            1. set to scan all sources
+  1. *[Checkstyle IDEA Plugin](https://github.com/jshiell/checkstyle-idea)*
+        1. After installation finish configuration in: **Settings > Tools > Checkstyle**
+            1. Load checkstyle file by clicking on the "plus" and navigating to the file .\IdeaProjects\triplea\config\checkstyle (If you can't find it, you can download it from [the repository](https://github.com/triplea-game/triplea/blob/master/config/checkstyle/checkstyle.xml))
+            1. Set checkstyle version
+            1. Set to scan all sources
       ![Screenshot from 2020-10-18 19-18-46
       ](https://user-images.githubusercontent.com/12397753/96394543-271e2700-1177-11eb-9460-24e2e235d60d.png)
+  1. *'final' for generated Variables (optional)*
+        1. **Settings > Editor > Code Style > Java** 
+        2. Go to tab "Code Generation" subsection "Final Modifier"
+            1. Check "Make generated local variables final"
+            2. Check "Make generated parameters final"
+![Screenshot_IntelliJ_MakeFinal_Generated](https://user-images.githubusercontent.com/10353640/147354665-a2a96231-86f8-4b6f-97b0-e4c6f729cdd0.png)
   1. *Save Actions (optional)*
-        1. **Settings > Other settings > Save Actions**
-        1. Select 'Activate save actions on save'
-        1. configure in settings to add 'final' to local variables and class variables.
+        1. **Settings > Editor > Inspections**
+            1. **Java > Code style issues**
+            2. Check "Local variable or parameter can be final"
+![Screenshot_IntelliJ_MakeFinal](https://user-images.githubusercontent.com/10353640/147356129-3f3e4877-43de-49a7-ab5b-73c4363f8b45.png)
+        2. **Settings > Tool > Actions on Save**
+            1. Check "Reformat code" (adapt to be only affected on "Changed lines")
+            2. Check "Optimize imports"
+            3. Check "Run code cleanup"
```

Comment:
I don't believe I have any of these turned anymore. The reformat code can fight with me while coding, optimize imports can remove imports that I'm about to use, and the code cleanup does not do a lot necessary. 

We have removed the checkstyle to require 'final' on all things where we can put it. Instead the guidance is to try and keep class variables final and immutable, and otherwise use 'final' where it makes sense. Long story short, perhaps all of the setup around automatically marking variables final and applying final can be removed.